index.js

```diff
@@ -1061,12 +1061,12 @@ Jimp.prototype.autocrop = function() {
  * @returns this for chaining of methods
 */
 Jimp.prototype.blit = function (src, x, y, srcx, srcy, srcw, srch, cb) {
-    if ("object" != typeof src || src.constructor != Jimp)
+    if (!(src instanceof Jimp))
         return throwError.call(this, "The source must be a Jimp image", cb);
     if ("number" != typeof x || "number" != typeof y)
         return throwError.call(this, "x and y must be numbers", cb);
 
-    if ("function" == typeof srcx) {
+    if (srcx instanceof Function) {
```

Comment:
The only case I can think of where `instanceof` check would fail is if `srcx` was defined in a different context (different frame or window). I think there's a chance of something like this happening when using browser version, but I've never used it myself and can't tell for sure.

I'd stick with `typeof` check personally